tests/unit/vendor/serverlessrepo/test_publish.py

```diff
@@ -222,6 +225,10 @@ def test_publish_existing_application_should_update_application_if_version_exist
         self.serverlessrepo_mock.create_application.assert_called_once()
         self.serverlessrepo_mock.update_application.assert_called_once()
         self.serverlessrepo_mock.create_application_version.assert_called_once()
+        # If --fail-on-same-version is specified, warning is printed
```

Comment:
This is the inverse right? If its not specified, we should expect to see a warning, but if we wanted to test that it publishes anyways, should we instead test if the `_check_app_with_semantic_version_exists()` method was not called, instead of testing log output?